synapse/app/synchrotron.py

```diff
@@ -371,16 +371,22 @@ def get_streams_to_replicate(self):
     def get_currently_syncing_users(self):
         return self.presence_handler.get_currently_syncing_users()
 
-    @defer.inlineCallbacks
-    def process_and_notify(self, stream_name, token, rows):
+    async def process_and_notify(self, stream_name, token, rows):
         try:
             if stream_name == "events":
                 # We shouldn't get multiple rows per token for events stream, so
                 # we don't need to optimise this for multiple rows.
                 for row in rows:
                     if row.type != EventsStreamEventRow.TypeId:
                         continue
-                    event = yield self.store.get_event(row.data.event_id)
+                    assert isinstance(row, EventsStreamRow)
+
+                    event = await self.store.get_event(
+                        row.data.event_id, allow_rejected=True
+                    )
```

Comment:
Do we want to add handling/logging if we can't find an event while we're here?

Reply:
not being able to find an event seems a sufficient edge-case that the existing logging is probably sufficient?

Reply:
Oh right, yes, sorry thought we weren't catching exceptions.